enterprise/coderd/schedule/template_test.go

```diff
@@ -606,6 +606,95 @@ func TestTemplateUpdateBuildDeadlinesSkip(t *testing.T) {
+			user  = dbgen.User(t, db, database.User{})
+			file  = dbgen.File(t, db, database.File{
+				CreatedBy: user.ID,
+			})
+			templateJob = dbgen.ProvisionerJob(t, db, nil, database.ProvisionerJob{
+				FileID:      file.ID,
+				InitiatorID: user.ID,
+				Tags: database.StringMap{
+					"foo": "bar",
+				},
+			})
+			timeTilDormant  = time.Minute * 2
+			templateVersion = dbgen.TemplateVersion(t, db, database.TemplateVersion{
+				CreatedBy:      user.ID,
+				JobID:          templateJob.ID,
+				OrganizationID: templateJob.OrganizationID,
+			})
+			template = dbgen.Template(t, db, database.Template{
+				ActiveVersionID:          templateVersion.ID,
+				CreatedBy:                user.ID,
+				OrganizationID:           templateJob.OrganizationID,
+				TimeTilDormant:           int64(timeTilDormant),
+				TimeTilDormantAutoDelete: int64(timeTilDormant),
+			})
+		)
+
+		// Add two dormant workspaces and one active workspace.
+		dormantWorkspaces := []database.Workspace{
+			dbgen.Workspace(t, db, database.Workspace{
+				OwnerID:        user.ID,
+				TemplateID:     template.ID,
+				OrganizationID: templateJob.OrganizationID,
+				LastUsedAt:     time.Now().Add(-time.Hour),
+			}),
+			dbgen.Workspace(t, db, database.Workspace{
+				OwnerID:        user.ID,
+				TemplateID:     template.ID,
+				OrganizationID: templateJob.OrganizationID,
+				LastUsedAt:     time.Now().Add(-time.Hour),
+			}),
+		}
+		dbgen.Workspace(t, db, database.Workspace{
+			OwnerID:        user.ID,
+			TemplateID:     template.ID,
+			OrganizationID: templateJob.OrganizationID,
+			LastUsedAt:     time.Now(),
+		})
+		for _, ws := range dormantWorkspaces {
+			db.UpdateWorkspaceDormantDeletingAt(ctx, database.UpdateWorkspaceDormantDeletingAtParams{
+				ID: ws.ID,
+				DormantAt: sql.NullTime{
+					Time:  ws.LastUsedAt.Add(timeTilDormant),
+					Valid: true,
+				},
+			})
+		}
+
+		// Setup dependencies
+		notifyEnq := testutil.FakeNotificationsEnqueuer{}
+		logger := slogtest.Make(t, &slogtest.Options{IgnoreErrors: true}).Leveled(slog.LevelDebug)
+		const userQuietHoursSchedule = "CRON_TZ=UTC 0 0 * * *" // midnight UTC
+		userQuietHoursStore, err := schedule.NewEnterpriseUserQuietHoursScheduleStore(userQuietHoursSchedule, true)
+		require.NoError(t, err)
+		userQuietHoursStorePtr := &atomic.Pointer[agplschedule.UserQuietHoursScheduleStore]{}
+		userQuietHoursStorePtr.Store(&userQuietHoursStore)
+		templateScheduleStore := schedule.NewEnterpriseTemplateScheduleStore(userQuietHoursStorePtr, &notifyEnq, logger)
+		templateScheduleStore.TimeNowFn = time.Now
+
+		// Update dormancy TTL for a lower value
+		_, err = templateScheduleStore.Set(ctx, db, template, agplschedule.TemplateScheduleOptions{
+			TimeTilDormant:           timeTilDormant / 2,
+			TimeTilDormantAutoDelete: timeTilDormant / 2,
+		})
+		require.NoError(t, err)
+
+		// We should expect two notifications. One for each dormant workspace.
+		require.Len(t, notifyEnq.Sent, len(dormantWorkspaces))
```

Comment:
Can we be more specific here?
Which notification is being received? What are its labels? etc.
If you had this you would've caught that your new notification was never being sent.

Reply:
+1